types/coin.go

```diff
@@ -599,15 +599,33 @@ var (
 	reAmt       = `[[:digit:]]+`
 	reDecAmt    = `[[:digit:]]*\.[[:digit:]]+`
 	reSpc       = `[[:space:]]*`
-	reDnm       = regexp.MustCompile(fmt.Sprintf(`^%s$`, reDnmString))
-	reCoin      = regexp.MustCompile(fmt.Sprintf(`^(%s)%s(%s)$`, reAmt, reSpc, reDnmString))
-	reDecCoin   = regexp.MustCompile(fmt.Sprintf(`^(%s)%s(%s)$`, reDecAmt, reSpc, reDnmString))
+	reDnm       = returnReDnm
+	reCoin      = returnReCoin
+	reDecCoin   = returnDecCoin
 )
 
+func returnDecCoin() *regexp.Regexp {
```

Comment:
And so if I write the global `CoinDenomRegex` it will let me override all these checks done inside of Coin.

Reply:
Exactly

Reply:
https://github.com/interchainberlin/pooltoy/blob/master/cmd/pooltoyd/main.go#L37